include/openspace/rendering/renderable.h

```diff
@@ -141,8 +141,26 @@ class Renderable : public properties::PropertyOwner, public Fadeable {
     // An optional renderbin that renderables can use for certain components, in cases
     // where all parts of the renderable should not be rendered in the same bin
     std::optional<RenderBin> _secondaryRenderBin;
-private:
 
+    struct AlternativeTransform {
+        std::optional<glm::dvec3> translation = std::nullopt;
+        std::optional<glm::dmat3> rotation = std::nullopt;
+        std::optional<glm::dvec3> scale = std::nullopt;
+    };
+
+    glm::dmat4 calcModelTransform(const RenderData& data,
```

Comment:
Just a reminder that we should try to add doxygen comments, especially for public methods